File: cpp_trainings/src/range_sensor_sub.cpp
#include "rclcpp/rclcpp.hpp"
#include "tutorial_interfaces/msg/range_sensor.hpp"
#include <time.h>
 
class RangeSensorSub : public rclcpp::Node 
{
public:
    RangeSensorSub() : Node("range_sensor_sub") 
    {
        RCLCPP_INFO(this->get_logger(), "Range sensor publisher node has been started..");
        subscription_ = this->create_subscription<tutorial_interfaces::msg::RangeSensor>("range_sensor",10, 
            std::bind(&RangeSensorSub::callbackTopic,this, std::placeholders::_1));
    }
 
private:
    void callbackTopic(const tutorial_interfaces::msg::RangeSensor::SharedPtr range_value)
    {
        RCLCPP_INFO(this->get_logger(),"At %d time range sensor value is %d", this->get_clock()->now(), range_value->range);
    }
    rclcpp::Subscription<tutorial_interfaces::msg::RangeSensor>::SharedPtr subscription_ ;
};
 
int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<RangeSensorSub>(); 
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: python_trainings/python_trainings/range_sensor_sub.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from tutorial_interfaces.msg import RangeSensor
 
 
class RangeSensorSub(Node): 
    def __init__(self):
        super().__init__("range_sensor_sub") 
        self.subs_ = self.create_subscription(RangeSensor, "RangeSensor", self.callback_message, 10)
    
    def callback_message(self,msg):
        self.get_logger().info('The range sensor value is : "%s"' % str(msg.range))
 
 
def main(args=None):
    rclpy.init(args=args)
    node = RangeSensorSub() 
    rclpy.spin(node)
    rclpy.shutdown()
 
 
if __name__ == "__main__":
    main()